pkg/connector/source.go

```diff
@@ -227,14 +227,16 @@ func (s *source) Read(ctx context.Context) (record.Record, error) {
 		return r, err
 	}
 
+	// TODO rethink if there's an actual benefit in setting these fields
```

Comment:
It would be cleaner and "more correct" to leave the field `nil`, on the other hand it would mean we have to check for `nil` every time we access a key or payload field, even just calling `Bytes()` on it would panic.

Another option would be to introduce a type `NilData` that just does nothing, then it wouldn't look like it's empty raw data.